### DISCUSSION TITLE
Triggering ListView.Selected

### DISCUSSION THREAD

**Discussion by Gabe-Pankoski** [Textualize/textual#1567] [Q&A] [answered]
For whatever reason, neither ListView.Highlight or Listview.Selected are calling handlers. I am working on Macos 13.0.1, with Python 3.11

```Python
class fileReader(App):
    files = reactive([])

    def compose(self) -> ComposeResult:
        for file in os.listdir():
            if file[0] != ".":
                self.files.append(file)
                logging.info(f"Found file: {file}")

        yield Header()
        yield Vertical(
                ListView(
                    *[ListItem( Label(f"{file}"), id=f"{file}" ) for file in self.files]
                ),
               TextLog(name="log", wrap=True, classes="box")
        )

    def on_listview_highlighted(self, event: ListView.Highlighted) -> None:
        itemID = event.ListView.item.id
        logging.info(f"Selected: {itemID}")
```

Unless I am doing something incredibly wrong, I don't understand why it shouldn't be working.

**Comment by davep** [comment, marked as answer]
The event handler for the ListView highlight message should be `on_list_view_highlighted` -- note the `_` in `list_view`. [One of the rules for the handler name that is looked for is CamelCase gets turned into snake_case](https://textual.textualize.io/guide/events/#handler-naming).

Here's a little standalone variation on what you've got above that will show the events in a `TextLog`:

```python
import os

from textual.app        import App, ComposeResult
from textual.containers import Horizontal
from textual.widgets    import Header, Footer, ListView, ListItem, TextLog, Label
from textual.reactive   import reactive

class FileReader( App[ None ] ):

    CSS = """
    ListView {
        width: 1fr;
    }
    TextLog {
        width: 1fr;
    }
    """

    files = reactive([])

    def compose( self ) -> ComposeResult:
        for file in os.listdir():
            if file[0] != ".":
                self.files.append(file)
        yield Header()
        yield Horizontal(
            ListView( *[
                ListItem( Label( f"{file}") , id=f"{file}" )
                for file in self.files
            ] ),
            TextLog()
        )
        yield Footer()

    def on_list_view_highlighted( self, event: ListView.Highlighted ) -> None:
        self.query_one( TextLog ).write( event )

    def on_list_view_selected( self, event: ListView.Selected ) -> None:
        self.query_one( TextLog ).write(event )

if __name__ == "__main__":
    FileReader().run()
```

**Reply by Gabe-Pankoski** [reply]
Wow, I feel a bit silly now. For some reason I was reading listitem and listview as one word as opposed to separating them for the handler. This will be good to know, thank you.

**Reply by davep** [reply]
Nah, don't, it's a very easy mistake to make -- done it a couple of times myself and sat wondering why my code isn't doing what I intended. I've seen it catch folk out before [and it's caused me to raise an issue for next week to perhaps try and make it a bit more clear in the docs](https://github.com/Textualize/textual/issues/1568) so you asking has helped improve things. :-)
